### ISSUE TITLE
[Module Proposal]: `avm/res/web/site`

### ISSUE THREAD

**Issue by tsc-buddy** [Azure/Azure-Verified-Modules#172]
### Check for previous/existing GitHub issues/module proposals

- [X] I have checked for previous/existing GitHub issues/module proposals

### Check this module doesn't already exist in another project that is being evolved into AVM

- [X] I have checked for that this module doesn't already exist in CARML or TFVM

### Terraform or Bicep?

Bicep

### Module Classification?

Resource Module

### Module Name

avm/res/web/site

### Module Details

After discussing with Mate Barabas, I will pick up the CARML conversion for App Service, App Service Plan, App Service Environment.

### Do you want to be the owner of this module?

Yes

### Module Owner's GitHub Username

tsc-buddy

**Comment by tsc-buddy** [comment]
@matebarabas - I also have a colleague who is happy to co-own these. So when we sync next, let me know the process to onboard them.

**Comment by prjelesi** [comment]
@matebarabas i will handle this together with other conversion request. @tsc-buddy i will share with you upgrade checklist that you can use to continuo your work.

**Comment by matebarabas** [comment]
@tsc-buddy, thanks for opening this request and volunteering! I'll shortly send you a request to officially confirm ownership. 

Also, @prjelesi is going to reach out to you separately and help with the next steps, given this is a migrate from CARML scenario. 

> @matebarabas - I also have a colleague who is happy to co-own these. So, when we sync next, let me know the process to onboard them.

Please let me know who the other person would be; we can capture him/her as a secondary owner.

Thank you!

**Comment by matebarabas** [comment]
Hi @tsc-buddy ,

Thanks for requesting/proposing to be an AVM module owner.

We just want to confirm you agree to the below pages that define what module ownership means:

- https://azure.github.io/Azure-Verified-Modules/specs/shared/team-definitions
- https://azure.github.io/Azure-Verified-Modules/specs/shared
- https://azure.github.io/Azure-Verified-Modules/help-support/module-support

Any questions or clarifications needed, let us know.

If you agree please just reply to this issue with the exact sentence below (as this helps with our automation 👍):

"I CONFIRM I WISH TO OWN THIS AVM MODULE AND UNDERSTAND THE REQUIREMENTS AND DEFINITION OF A MODULE OWNER"

Thanks,

The AVM Core Team

**Comment by tsc-buddy** [comment]
"I CONFIRM I WISH TO OWN THIS AVM MODULE AND UNDERSTAND THE REQUIREMENTS AND DEFINITION OF A MODULE OWNER"

**Comment by matebarabas** [comment]
Hi @tsc-buddy,

Thanks for confirming that you wish to own this AVM module and understand the related requirements and responsibilities!

We just want to ask you to double check a few important things before you start the development.

**Please use the following values explicitly as provided in the [module index](https://azure.github.io/Azure-Verified-Modules/indexes/) page**:

- For your module:
  - `ModuleName` - to name your module
  - `TelemetryIdPrefix` - to be used in your module's telemetry
- For your module's repository:
  - Repository name and folder path are defined in `RepoURL`
  - Create the GitHub teams defined in the `ModuleOwnersGHTeam` and `ModuleContributorsGHTeam` columns

You can now start the development of this module! ✅ Happy coding! 🎉

**Please respond to this comment, once your module is ready to be published!** 🙏

Any further questions or clarifications needed, let us know!

Thanks,

The AVM Core Team

<br>

> **NOTE**:
>
> - This Module Proposal issue **MUST remain open** until the module is fully developed, tested and published to the relevant registry. **Do NOT close** the issue before the successful publication is confirmed!
> - Once the module is fully developed, tested and published to the relevant registry, and the Module Proposal issue was closed, it **MUST remain closed**.

**Comment by matebarabas** [comment]
Hi @tsc-buddy, I'm reaching out as having this issue revisited, I now believe that this one might have been given the wrong name. 

If I'm not mistaken, the intent here is to cover Web Apps and Function apps (as these are the assets deployed into an App Service Plan. The correct name for this one is `avm-res-web-site`, and the donor module in CARML is here: https://github.com/Azure/ResourceModules/tree/main/modules/web/site

**Can you please confirm that this GH issue is meant to cover this module?**

> Note: I don't want to further confuse you, but please expect the naming convention for Bicep modules to slightly change. I'll reach out to you once again when this happens.

**Comment by matebarabas** [comment]
@tsc-buddy,

Due to a recent change in the naming convention for AVM Bicep modules, this module is getting an updated name.

The title and description of this GH issue has been updated to reflect the name change. **Please make sure you adhere to the new naming convention when publishing your module!** 

Please see the new naming convention in:
- [RMNFR1](https://azure.github.io/Azure-Verified-Modules/specs/shared/#bicep-resource-module-naming) for resource modules
- [PMNFR1](https://azure.github.io/Azure-Verified-Modules/specs/shared/#bicep-pattern-module-naming) for pattern modules

Please note that the Telemetry ID and name of GitHub Teams haven't changed.

If you feel uncertain about any details, please check the [Bicep modules indexes](https://azure.github.io/Azure-Verified-Modules/indexes/bicep/).

**Comment by matebarabas** [comment]
- To be migrated from CARML: https://github.com/Azure/ResourceModules/tree/main/modules/web/site
- Requested by the AZD team: #261 
CC: @jongio

**Comment by krbar** [comment]
@matebarabas FYI: I will support @tsc-buddy with the migration of this module as I need it for a customer project.

**Comment by matebarabas** [comment]
(draft) PR: https://github.com/Azure/bicep-registry-modules/pull/779

**Comment by krbar** [comment]
@matebarabas @tsc-buddy PR has been merged into the BRM repo: https://github.com/Azure/bicep-registry-modules/pull/779

**Comment by matebarabas** [comment]
- [x] PR merged; Module published. 
- [x] GitHub teams are available and assigned to their parents.
- [x] `CODEOWNERS` file updated.
 
Thank you for your work @tsc-buddy and @krbar! I'm closing this issue now.
